### PR TITLE
OAuth fixes

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
I've recently setup OAuth2 for an MS Exchange IMAP mail account for the email to activity processor. Have come across a few problems...

System: CiviCRM 5.51.1 on WordPress and PHP7.4.

The following issues are fixed in this PR:
- Authentication fails when scheduled job is run via CRON/CLI with `invalid_client` because the redirectURL is being generated as a frontend URL. When run via the scheduled job UI "Execute Now" it generates a backend URL and that's what is shown/configured in OAuth configuration UI.
- OAuthException not defined - should be \Civi\OAuth\OAuthException.
- Replace deprecated getTimeRaw() with time().
- Code cleanup / code standards.

Note that this does NOT fix https://lab.civicrm.org/dev/core/-/issues/2559 which is a blocker to using MS Exchange OAuth2 without editing core files.

Before
----------------------------------------
Various issues.

After
----------------------------------------
Fixed.

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by demeritcowboy** [comment]
I guess you're the first person to use it with wordpress!?

These all look ok to me just maybe looking for a 2nd opinion on the access check one, although it should fail either way because without an access token how can you get a refresh token? I'll give it a little run.

File: ext/oauth-client/Civi/Api4/Action/OAuthSysToken/Refresh.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ protected function getSelect() {
   }
 
   protected function doTask($row) {
-    if ($this->threshold >= 0 && \CRM_Utils_Time::getTimeRaw() < $row['expires'] - $this->threshold) {
+    if (!empty($token['access_token']) && ($this->threshold >= 0 && \CRM_Utils_Time::time() < $row['expires'] - $this->threshold)) {
```
Comment (demeritcowboy) [inline ext/oauth-client/Civi/Api4/Action/OAuthSysToken/Refresh.php:53]: You must have meant `$row['access_token']`.

It runs ok in various scenarios with $row, but would still be looking for a second opinion as to what it's supposed to do if it's empty, and I'm wondering how it helps troubleshooting.

Comment (mattwire) [reply ext/oauth-client/Civi/Api4/Action/OAuthSysToken/Refresh.php]: I've removed it from the PR :-)

**Comment by mattwire** [comment]
> These all look ok to me just maybe looking for a 2nd opinion on the access check one, although it should fail either way because without an access token how can you get a refresh token?

Just noting that the check for empty access token has now been removed from the PR

**Comment by mattwire** [comment]
Thanks @demeritcowboy

**Pull request merged**